### PR TITLE
Orca:  Polish Resnet50 Inference Example

### PR DESCRIPTION
**Pull request by leonardozcm**
## Description

https://github.com/intel-analytics/BigDL/issues/6955

### CODE REVIEW

**Comment by hkvision** [comment]
Need to add this to example tests

File: .github/actions/orca/orca-exampletest-ray-py37-spark3-action/pr-test/action.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+runs:
+  using: "composite"
+  steps:
+    - name: Run Test
+      shell: bash
+      run: |
+        source activate py37
+        export SPARK_LOCAL_HOSTNAME=localhost
+        
+        bash python/dev/release_default_linux_spark313.sh default false false false -Ddata-store-url=$HTTP_URI -U
```
Comment (hkvision) [inline .github/actions/orca/orca-exampletest-ray-py37-spark3-action/pr-test/action.yml:12]: spark3

File: .github/actions/orca/orca-exampletest-ray-py37-spark3-action/pr-test/action.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+name: 'Run Orca Python Py37 Spark3'
+description: 'Run Orca Python Py37 Spark3'
```
Comment (hkvision) [inline .github/actions/orca/orca-exampletest-ray-py37-spark3-action/pr-test/action.yml:2]: modify the name

**Comment by leonardozcm** [comment]
tested on imagenet-small:
```python
total number of records: 2048
batch_size for each worker: 256
num_samples for each worker: around 1024.0
num_batches for each worker: around 4.0
avg_val_time for each worker: 183.85887551307678
avg_forward_time for each batch: 22.529457822442055
inference latency 88.006 ms
Throughput: 11.363 fps
Accuracy: 0.812 
Stopping orca context
```

File: .github/actions/orca/orca-exampletest-ray-py37-spark3-action/pr-test/action.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        pip install -i https://pypi.org/simple python/orca/src/dist/bigdl_orca*-py3-none-manylinux1_x86_64.whl
+
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} tensorflow==1.15.0
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} tensorflow-gan==2.0.0
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} tensorflow-probability==0.7.0
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} tensorflow-datasets==3.2.0
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} torch==1.7.1
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} torchvision==0.8.2
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} torchmetrics==0.10.0
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} ray[default]==1.9.2
```
Comment (hkvision) [inline .github/actions/orca/orca-exampletest-ray-py37-spark3-action/pr-test/action.yml:27]: use pip install -u requirements_ray.text (refer to other jobs)

File: python/orca/example/learn/pytorch/resnet50/inference.py [hunk 4]
```diff
@@ -285,7 +258,6 @@ def model_creator(config):
                     model = torch.jit.trace(model, x)
                     model = torch.jit.freeze(model.eval())
                     y = model(x)
-                    y = model(x)
```
Comment (hkvision) [inline python/orca/example/learn/pytorch/resnet50/inference.py:288]: may not remove this? Original code has this, may need to forward twice intentionally.

File: python/orca/example/learn/pytorch/resnet50/inference.py [hunk 5]
```diff
@@ -99,87 +92,67 @@ def validate(self, val_iterator, info, metrics, num_steps=None):
...
+
+    def on_val_forward(self, runner):
+        if runner.config["dummy"]:
+            images = torch.randn(runner.config["batch"], 3, 224, 224)
+            target = torch.arange(1, runner.config["batch"] + 1).long()
+            # Only do the conversion once for dummy data
+            if runner.config["ipex"]:
+                images = images.contiguous(memory_format=torch.channels_last)
+            if runner.config["bf16"]:
+                images = images.to(torch.bfloat16)
```
Comment (hkvision) [inline python/orca/example/learn/pytorch/resnet50/inference.py:121]: This should be done only once but not for each iteration. Re-check the logic of the original code?

File: python/orca/example/learn/pytorch/resnet50/inference.py [hunk 6]
```diff
@@ -99,87 +92,67 @@ def validate(self, val_iterator, info, metrics, num_steps=None):
...
+            output, target, loss = self.forward(runner, images, target, warmup=True)
+        else:
+            output, target, loss = self.forward(runner, images, target, warmup=False)
 
-        if self.use_tqdm and self.world_rank == 0:
-            from tqdm import tqdm
-            _progress_bar = tqdm(
-                total=num_steps,
-                unit="batch",
-                leave=False)
```
Comment (hkvision) [inline python/orca/example/learn/pytorch/resnet50/inference.py:123]: This part is already included in TqdmCallback right?

Comment (leonardozcm) [reply python/orca/example/learn/pytorch/resnet50/inference.py:123]: Yes

**Pull request merged**